[Bug]: v7: "global" module alias breaks package that uses the real "global" package

### Describe the bug

Storybook crashes with

```
ModuleNotFoundError: Module not found: Error: Can't resolve 'global/window' in 'xhr'
```

[global](https://www.npmjs.com/package/global) is its own package, and is used by a package [xhr](https://www.npmjs.com/package/xhr).

![image](https://user-images.githubusercontent.com/5390719/221146930-610aa609-3649-46e0-96e4-2e2976bbc912.png)

storybook v7's default webpack config includes an alias which breaks the build.

### To Reproduce

_No response_

### System

```shell
Environment Info:

  System:
    OS: Windows 10
  Binaries:
    Node: 19.3.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.2.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1325.0), Chromium (110.0.1587.50)
  npmPackages:
    @storybook/addon-actions: ^7.0.0-beta.53 => 7.0.0-beta.53
    @storybook/addon-essentials: ^7.0.0-beta.53 => 7.0.0-beta.53
    @storybook/addon-links: ^7.0.0-beta.53 => 7.0.0-beta.53
    @storybook/react: ^7.0.0-beta.53 => 7.0.0-beta.53
    @storybook/react-webpack5: ^7.0.0-beta.53 => 7.0.0-beta.53
```


### Additional context

_No response_

--- a/code/lib/builder-webpack5/src/preview/iframe-webpack.config.ts
+++ b/code/lib/builder-webpack5/src/preview/iframe-webpack.config.ts
@@ -30,7 +30,6 @@ import { createBabelLoader } from './babel-loader-preview';
 const wrapForPnP = (input: string) => dirname(require.resolve(join(input, 'package.json')));
 
 const storybookPaths: Record<string, string> = {
-  global: wrapForPnP('@storybook/global'),
   ...[
     // these packages are not pre-bundled because of react dependencies
     'api',

